types/share_merging.go

```diff
@@ -26,10 +26,12 @@ func DataFromSquare(eds *rsmt2d.ExtendedDataSquare) (Data, error) {
 
 	// iterate over each row index
 	for x := uint(0); x < originalWidth; x++ {
-		// iterate over each col index
-		for y := uint(0); y < originalWidth; y++ {
+		// iterate over each share in the original data square
+		row := eds.Row(x)
+		originalWidth := eds.Width() / 2
+
+		for _, share := range row[:originalWidth] {
```

Comment:
Here's the update to the latest rsmt2d api, which is the most meaningful change. I was careful not to change anything else.